src/System.Text.Json/tests/Serialization/Value.ReadTests.cs

```diff
@@ -297,9 +297,15 @@ public static void ReadPrimitiveUri()
         {
             Uri uri = JsonSerializer.Deserialize<Uri>(@"""https://domain/path""");
             Assert.Equal("https:\u002f\u002fdomain\u002fpath", uri.ToString());
+            Assert.Equal("https://domain/path", uri.OriginalString);
```

Comment:
Do we have tests for both relative and absolute paths, along with those that end up escaping? If not, we should add some.

Particularly, cases that were mentioned here:
https://faithlife.codes/blog/2010/08/uritostring_must_die/
https://coding.abel.nu/2014/10/beware-of-uri-tostring/

Thanks @bartonjs for example links of why using `ToString()` may not be appropriate here.

Reply:
I did have one, I added a second.